src/GdbServer.cc

```diff
@@ -1396,10 +1396,9 @@ static uint32_t get_cpu_features(SupportedArch arch) {
   return cpu_features;
 }
 
-static const char localhost_addr[] = "127.0.0.1";
-
 struct DebuggerParams {
   char exe_image[PATH_MAX];
+  char host[16]; // INET_ADDRSTRLEN, omitted for header churn
```

Comment:
Let's make this a C++ `string`.

Reply:
I think this should remain as-is, but I am open to your thoughts.

DebuggerParams is a C struct so that this information can be serialized and sent over a socket via `read`/`write` calls for parent/child communication.

https://github.com/mozilla/rr/blob/6b65cd4c72a033f7b11d765a35d8864a9a45cd09/src/GdbServer.cc#L1511
https://github.com/mozilla/rr/blob/6b65cd4c72a033f7b11d765a35d8864a9a45cd09/src/GdbServer.cc#L1587

Changing this to an std::string means we'll be sending the shell of a C++ object over the socket rather than the underlying hostname itself.

Reply:
You're quite right, sorry!